src/data/markdown/docs/02 javascript api/11 jslib/01 httpx.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "httpx"
+excerpt: "httpx is a wrapper library around the native k6 http module"
+---
+
+The `httpx` module is an external JavaScript library that wraps around the native [k6/http](/javascript-api/k6-http) module. 
+It's a http client with features that are not yet available in the native module.
+ - ability to set set http options globally (such as timeout)
+ - ability to set default tags and headers that will be used for all requests
+ - more user-friendly arguments to request functions (get, post, put take the same arguments)
+
+
+
+httpx module integrates well with functional library. 
+
+
+<Blockquote mod='warning'>
+
+#### This library is in active development
+
+This library is stable enough to be useful, but pay attention to the new versions released in jslib.k6.io. 
+
+This documentation is for the last version only. If you discover that some of the code below does not work, it most likely means that you are using an older version.
+
+</Blockquote>
+
+
+### Example
+
+<CodeGroup labels={["httpx session used together with functional library"]}>
+
+```javascript
+import { test } from 'https://jslib.k6.io/functional/0.0.1/index.js';
+import { Httpx } from 'https://jslib.k6.io/httpx/0.0.3/index.js';
+import { randomIntBetween, 
+         randomItem } from "https://jslib.k6.io/k6-utils/1.0.0/index.js";
+
+const USERNAME = `user${randomIntBetween(1, 100000)}@example.com`;  // random email address
+const PASSWORD = 'superCroc2021';
+
+let session = new Httpx({
+    baseURL: 'https://test-api.k6.io', 
+    headers: {
+        'User-Agent': "My custom user agent",
+        "Content-Type": 'application/x-www-form-urlencoded' 
+    },
+    timeout: 20000 // 20s timeout.
+});
+
+export default function testSuite() {
+
+  test(`Create a test user ${USERNAME}`, (t) => {
+
+    let resp = session.post(`/user/register/`, {
+      first_name: 'Crocodile',
+      last_name: 'Owner',
+      username: USERNAME,
+      password: PASSWORD,
+    });
+
+    t.expect(resp.status).as("status").toEqual(201)
+      .and(resp).toHaveValidJson();
+  })
+
+  &&
```

Comment:
maybe a comment that this is important ?

Reply:
Why is it that `&&` is even needed? I would expect this to become the most common source of errors for users of this lib.

Reply:
Fair point, it either needs to be documented or changed. There's a good reason for having `&&` but maybe this coding style is too confusing.

Reply:
I think so. Adding a way to chain them using fluent syntax would probably be less confusing if you still want the `&&` behavior? Like `test().and.test().and.test().and.test()`?

Reply:
> coding style is too confusing
I think that too.